containerz/containerz.proto

```diff
@@ -64,6 +80,24 @@ service Containerz {
   // (i.e. kill) a container.
   rpc StopContainer(StopContainerRequest) returns (StopContainerResponse) {}
 
+  // UpdateContainer updates a running container to the image specified in the
+  // request. By default the operation is synchronous which means that the
+  // request will only return once the container has either been successfully
+  // updated or the update has failed. If the client requests an asynchronous
+  // update then the server must perform all validations (e.g. does the
+  // requested image exist on the system or does the instance name exist) and
+  // return to the client and the update happens asynchronously. If the update
+  // fails, the server must restore the previous version of the container. It
+  // is up to the client to check if the update actually updates the container
+  // to the requested version or not.
+  // If no StartContainerRequest is provided, the server must reuse the same
+  // run specification as the existing running containers. Otherwise, it must
+  // use the provided StartContainerRequest message.
+  // If a container exists but is not running then this operation should return
+  // an error.
```

Comment:
If a container is failing due to vendor OS upgrade i.e. incompatibility b/w application and OS, the best way to fix this is by rolling out the fix forward. So we should allow the upgrade to happen even if the old container is not running.

Reply:
Ahh good point - I changed it such that an upgrade should always happen.